internal/command/machine/run.go

```diff
@@ -659,8 +667,13 @@ func determineMachineConfig(ctx context.Context, initialMachineConf api.MachineC
 		machineConf.Init.Entrypoint = splitted
 	}
 
-	if cmd := flag.Args(ctx)[1:]; len(cmd) > 0 {
-		machineConf.Init.Cmd = cmd
+	if cmd := flag.Args(ctx)[1:]; len(cmd) > 0 && len(machineConf.Init.Cmd) == 0 {
+		cmd := strings.Join(cmd, " ")
+		split, err := shlex.Split(cmd)
```

Comment:
no need to join/split the args here, they were already splitted correctly by the shell when call flyctl.